Automattic/wp-calypso
# Improve unified mailbox list with several critical fixes

This set of changes build on #64766 by adding several fixes:
- Fixes flickering fields by ensuring components are no longer inlined.
- Moves state dependent logic into `useEffect` hooks for `<NewMailboxList/>`
- Makes it possible to switch of `autoFocus`
- Ensures that only unique mailbox names are allowed for submission

#### Testing Instructions



* Ensure you have a domain without an email subscription
* Click on the [ **+Add** ] from the domain management view i.e. [ **Upgrades** ] > [ **Domains** ] > [ + **Add**  ] . Alternatively navigate to `/email/<:domain>/purchase/<:siteSlug>`
* Append `?flags=unify-mailbox-forms` to the end of the URL to activate the `unify-mailbox-forms` feature flag.
* Confirm that no flickering occurs on the input fields
* Confirm that the form and validation works as usual.
* Confirm that duplicate mailbox names can't be supplied for the same form
* Confirm that you now see an option to add multiple mailboxes to the new subscription.
* Confirm that you can checkout successfully.



Related to #64766
